connectors/slack/template/index.js

```diff
@@ -1,24 +1,321 @@
+                cb(err, null);
+            }
+        );
+    } else {
+        if (ctx.configuration.slack_signing_secret) {
+            if (pathname.startsWith('/events/ingest') && ctx.method === 'POST') {
+                handleEventIngest(ctx, cb);
+            } else if (pathname.startsWith('/events/registration') && ctx.method === 'POST') {
+                return registerCore(ctx, cb);
+            } else if (pathname.startsWith('/events/registration') && ctx.method === 'DELETE') {
+                return registerCore(ctx, cb, true);
+                // COMMENT OUT WHEN NOT DEBUGGING
+                // } else if(pathname.startsWith('/events/storage') && ctx.method === 'GET') {
+                //     storage.ensureStorage(ctx);
+                //     storage.ensureCache().then(()=>{
+                //         cb(null, {
+                //             status: 200,
+                //             body: {
+                //                 tokenToTeam : storage.tokenToTeam,
+                //                 teamToHandler: storage.teamToHandler
+                //             }
+                //         });
+                //         return;
+                //     });
+                // } else if(pathname.startsWith('/events/storage') && ctx.method === 'DELETE'){
+                //     storage.ensureStorage(ctx);
+                //     storage.ensureCache().then(()=>{
+                //         storage.put(()=>{
+                //             storage.tokenToTeam = {};
+                //             storage.teamToHandler = {};
+                //         }).then(()=>{
+                //             cb(null, { status: 200 });
+                //             return;
+                //         });
+                //     });
+            } else {
+                return cb(null, { status: 404 });
+            }
+        } else {
+            return cb(null, {
+                status: 404,
+                body: 'Connector is not configured for Slack Events API',
+            });
+        }
+    }
+};
 
+function handleEventIngest(ctx, cb) {
+    debugTime('handleEventIngest');
+    debugTimeLog(
+        'handleEventIngest',
+        `starting handleEventIngest ${ctx.query.dispatch !== undefined ? '(dispatch) ' : '(ingest) '}for`,
+        ctx.body && ctx.body.event && ctx.body.event.text
+    );
 
-Completion - 302 redirect to returnTo URL with query params
-- [required] status (success or error)
-- [required] data (augmented settings from initialization, or new settings if initialization did not provide any)
-- [required if present on initialization] state (from initialization)
+    try {
+        verifyRequestSignature({
+            signingSecret: ctx.configuration.slack_signing_secret,
+            requestSignature: ctx.headers['x-slack-signature'],
+            requestTimestamp: parseInt(ctx.headers['x-slack-request-timestamp'], 10),
+            body: JSON.stringify(ctx.body).replace(/\//g, '\\/'),
```

Comment:
Add this comment:
```
// Slack escapes forward slashes in their body text as "\/"; this is stripped out during the JSON parsing.
// Add it back in so that the signature is calculated correctly.
```

Reply:
Fixed